src/lib/autoDetect.ts

```diff
@@ -0,0 +1,73 @@
+import { ExtensionContext, workspace, window, languages } from 'vscode';
+import { FileAssociations } from './types';
+const { extname } = require('path');
+
+export const initializeHubLAutoDetect = (context: ExtensionContext) => {
+  languages.setLanguageConfiguration('hubl', {
+    indentationRules: {
+      decreaseIndentPattern: /.*{%(.*?end).*%}*./,
+      increaseIndentPattern: /.*{%(?!.*end).*%}.*/,
+    },
+  });
+  const languageExtensions = {
+    '.html': 'html-hubl',
+    '.css': 'css-hubl',
+  };
+  const getHublPreference = context.workspaceState.get('USE_HUBL');
+
+  const filesConfig = workspace.getConfiguration('files');
+  const fileAssoc = filesConfig.get('associations') as FileAssociations;
+  context.workspaceState.update('USE_HUBL', null);
+
+  const showHublDetectedMessage = (extension: '.html' | '.css') => {
+    const hublDetectedMessage = (extension: '.html' | '.css') => {
+      return `It looks like this file contains HubL code. Would you like to use HubL for all ${extension} files in this project?`;
+    };
+    const hublDetectedYesButton = `Use HubL for ${extension} files`;
+    const hublDetectedNoButton = 'No';
+
+    window
+      .showInformationMessage(
+        hublDetectedMessage(extension),
+        hublDetectedYesButton,
+        hublDetectedNoButton
+      )
+      .then((selection) => {
+        if (selection === hublDetectedYesButton) {
+          updateWorkspaceFileAssociation(extension);
+        } else if (selection === hublDetectedNoButton) {
+          context.workspaceState.update('USE_HUBL', false);
+        }
+        handleTextDocumentOpen.dispose();
+      });
+  };
+
+  const updateWorkspaceFileAssociation = (extension: '.html' | '.css') => {
+    fileAssoc[`*${extension}`] = languageExtensions[extension];
+    filesConfig.update('associations', fileAssoc);
+  };
+
+  const handleTextDocumentOpen = workspace.onDidOpenTextDocument(() => {
```

Comment:
If we check on startup and `USE_HUBL` is already false, we may want to avoid applying the event listener